### PR TITLE
Nightly Tests: add Envoy Tests Stage

### PR DESCRIPTION
**Pull request by ianvernon**
Add stage to run `make tests-envoy` as part of Nightly tests.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by eloycoto** [comment]
Hi @ianvernon 

- Why we run the test into runtime box? Could we run that in a docker container? 
-  Could we test this in parallel with the unit test?

**Comment by ianvernon** [comment]
@eloycoto this requires Bazel - so, we'd need something with Bazel installed + an up-to-date cache from my understand - @jrajahalme is this correct?

This is merely mirroring what happens in the Bash tests currently, as we currently do not run `make tests-envoy` in the Ginkgo tests, which I view as a blocker to disabling the Bash-script based build because we'd be losing test-coverage. 

I tried running in parallel with the tests themselves (Runtime, K8s-1.6, K8s-1.8) and hit issues with concurrency involving Bazel, so I put it into its own stage. I'm highly concerned by how long the build is taking, though: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/783/execution/node/50/log/ took 27 minutes just to run the envoy tests!

**Comment by eloycoto** [comment]
Quick question, could we have another jenkins hook, like `test-envoy` or something similar? Not sure if we need to add envoy test per each build. 

My proposal: 

- `Test-envoy` hook. 
- Run the test on Nightly or master, I'm open to this :-)

**Comment by ianvernon** [comment]
Talked to @jrajahalme offline - we can run this as part of nightly. I'll make a change to the Nightly Jenkinsfile instead of having it in `ginkgo.Jenkinsfile`.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-nightly

**Comment by ianvernon** [comment]
test-nightly

File: test/run-envoy-unit-tests.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -e
+
+vagrant ssh runtime -c "cd /src; make tests-envoy"
```
Comment (eloycoto) [inline test/run-envoy-unit-tests.sh]: @ianvernon  I'm not sure that this works, with the new image the /src/ no longer exists

**Review by eloycoto — DISMISSED** [review]

File: test/run-envoy-unit-tests.sh (hunk 1 above)
Comment (ianvernon) [reply test/run-envoy-unit-tests.sh]: Good catch, I'll fix that.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-nightly

**Comment by ianvernon** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Comment by ianvernon** [comment]
test-nightly

**Review by aanm — APPROVED** [review]

**Pull request merged**